source/apps/whix/wifi_whix.c

```diff
@@ -2110,23 +2110,29 @@ static void whix_route(wifi_event_route_t *route)
 
 static void whix_common_config_to_monitor_queue(wifi_monitor_data_t *data, bool is_channel_util)
 {
+    wifi_global_param_t *global_param = get_wifidb_wifi_global_param();
+
     data->u.mon_stats_config.inst = wifi_app_inst_whix;
 
-    wifi_global_param_t *global_param = get_wifidb_wifi_global_param();
-    if ((global_param != NULL) && (global_param->whix_chutility_loginterval != 0) && (global_param->whix_log_interval != 0)) {
-        if (is_channel_util) {
-            data->u.mon_stats_config.interval_ms = (global_param->whix_chutility_loginterval) * 1000;
+    if (is_channel_util) {
+        if ((global_param != NULL) && (global_param->whix_chutility_loginterval != 0)) {
+
+            data->u.mon_stats_config.interval_ms = (global_param->whix_chutility_loginterval) *
+                1000;
         } else {
```

Comment:
This logic might have unintended consequences - eg. values will be updated if global_logic == NULL. Is that expected?
```suggestion
if (global_param == NULL)
    return;    
if (is_channel_util) {
    if (global_param->whix_chutility_loginterval != 0) {
        data->u.mon_stats_config.interval_ms = (global_param->whix_chutility_loginterval) * 1000;
    } else {
         data->u.mon_stats_config.interval_ms = CHAN_UTIL_INTERVAL_MS;
    }
    if (global_param->whix_log_interval != 0) {
        data->u.mon_stats_config.interval_ms = (global_param->whix_log_interval) * 1000;
    } else {
        data->u.mon_stats_config.interval_ms = TELEMETRY_UPDATE_INTERVAL_MS;
    }
}
```
Please consider above.
This way it is much more easy to understand and resistant to error.